subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorCompositeBuildIntegrationTest.groovy

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.workers.internal
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.workers.fixtures.WorkerExecutorFixture
+import spock.lang.Unroll
+
+class WorkerExecutorCompositeBuildIntegrationTest extends AbstractIntegrationSpec {
+    WorkerExecutorFixture fixture = new WorkerExecutorFixture(temporaryFolder)
+    def plugin = testDirectory.createDir("plugin")
+    def build1 = testDirectory.createDir("build1")
+
+    @Unroll
+    def "can use worker api with composite builds using #pluginId"() {
+        settingsFile << """
+            includeBuild "plugin"
+            includeBuild "build1"
+        """
+
+        withFileHelperInPluginBuild()
+        withLegacyRunnableAndTaskInPluginBuild()
+        withTypedWorkerPluginInPluginBuild()
+
+        build1.file("build.gradle") << """
+            buildscript {
+                repositories {
+                    ${jcenterRepository()}
+                }
+
+                dependencies {
+                    classpath "org.apache.commons:commons-math:2.2"
```

Comment:
It _is_ necessary since classloader caching would cause an identical classpath to be returned for both builds unless there is some difference.  However, it's not necessary that it actually be a real dependency.  I've changed this to just an un-managed file and eliminated the external dependency.